src/optimagic/visualization/history_plots.py

```diff
@@ -132,36 +127,69 @@ def criterion_plot(
         if max_evaluations is not None and len(history) > max_evaluations:
             history = history[:max_evaluations]
 
-        scatter_kws = {
-            "connectgaps": True,
-            "showlegend": not plot_multistart,
-        }
-
         _color = next(palette)
         if not isinstance(_color, str):
             msg = "highlight_palette needs to be a string or list of strings, but its "
             f"entry is of type {type(_color)}."
             raise TypeError(msg)
 
-        line_kws = {
-            "color": _color,
-        }
-
-        trace = go.Scatter(
+        line_data = LineData(
             x=np.arange(len(history)),
             y=history,
-            mode="lines",
+            color=_color,
             name="best result" if plot_multistart else _data["name"],
-            line=line_kws,
-            **scatter_kws,
+            show_legend=not plot_multistart,
+        )
+        lines.append(line_data)
+
+    # ==================================================================================
+    # Store backend agnostic plotting data and configuration
+
+    plot_data = CriterionPlotData(
+        lines=lines,
+        multistart_lines=multistart_lines,
+    )
+
+    plot_config = PlotConfig(
```

Comment:
To be consistent with `CriterionPlotData` this should probably be called `CriterionPlotConfig`; however, I am not sure whether we can recycle this for other plotting functions. Can you have a look at some other plotting functions and the plot config they require?

Reply:
I am not a big fan of the backend prefixes. We would have to add an entry to PlotConfig for each backend. I could rather see that we do

```python
backend_to_legend_config: dict[PlotBackend, dict[str, Any]] = {
   PlotBackend.PLOTLY: {"yanchor": "top", "xanchor": "right", "y": 0.95, "x": 0.95},
   PlotBackend.MATPLOTLIB: {"loc": "upper right"},
 }
 
plot_config = PlotConfig(
    legend=backend_to_legend_config[requested_backend]
     ...
)
```

But for this PR where we do not have any backends, I would just use the entry `legend`.